skrub/_table_vectorizer.py

```diff
@@ -146,6 +146,76 @@ def _replace_missing_in_cat_col(ser: pd.Series, value: str = "missing") -> pd.Se
     return ser
 
 
+def _parallel_on_columns(trans: TransformerMixin, cols: list[str]):
+    """
+    Assert whether we want to parallelize the transformer over
+    the columns or not. We only want to parallelize if the transformer
+    is "univariate" (i.e. it can be duplicated for each column).
+    """
+    return (
+        (not isinstance(trans, str))
+        and trans._get_tags().get("univariate", False)
+        and len(cols) > 1
+    )
+
```

Comment:
It seems we can combine `_merge_fitted_transformers` and `_merge_unfitted_transformers` routines:

```python
from itertools import chain

def _merge_transformers(
    transformers_ list[tuple[str, str | TransformerMixin, list[str]]],
    is_fitted: bool,
    transformer_to_input_indices: dict[str, list[int]],
) -> tuple[list[TransformerMixin], dict[str, list[int]] | None]:
    """Merge a list of split transformers.
    """
    new_transformers = []
    new_transformer_to_input_indices = {}
    base_names = pd.unique([name.split("_split_")[0] for name, _, _ in transformers_])
    
    for base_name in base_names:
        # merge all transformers with the same base name
        transformers, columns, names = [], [], []
        for name, trans, cols in transformers:
            if name.startswith(base_name):
                columns.extend(cols)
                transformers.append(trans)
                names.append(name)

        new_trans = transformers[0]
	if is_fitted:
            if isinstance(new_trans, TransformerMixin):
                new_trans = clone(new_trans)
	else:
            if len(transformers) > 1:
                # merge transformers
                new_trans = new_trans.__class__._merge(transformers)
            new_transformer_to_input_indices[base_name] = list(
                chain.from_iterable([
                    _transformer_to_input_indices[name] for name in names
                ])
            )
        new_transformers.append(base_name, new_trans, columns)

    return new_transformers, new_transformer_to_input_indices
```

and then call it like:
```python
    def _merge_univariate_transformers(self):
        # merge back self.transformers and self.transformers_
        check_is_fitted(self, attributes=["transformers_"])
        self.transformers = _merge_transformers(self.transformers, is_fitted=False)
        (
            self.transformers,
            self._transformer_to_input_indices,
        ) = _merge_transformers(
            self.transformers_, is_fitted=True, self._transformer_to_input_indices
        )
```

WDYT?